src/lib.rs

```diff
@@ -177,45 +183,8 @@ impl ChallengeResponse {
         Ok((session_url.to_string(), crs))
     }
 
-    fn new_session_request(&self, nonce: &Nonce) -> Result<reqwest::blocking::Response, Error> {
-        let u = self.new_session_request_url(nonce)?;
-
-        let r = self
-            .http_client
-            .post(u.as_str())
-            .header(reqwest::header::ACCEPT, CRS_MEDIA_TYPE)
-            .send()?;
-
-        Ok(r)
-    }
-
-    fn new_session_request_url(&self, nonce: &Nonce) -> Result<url::Url, Error> {
-        let base = &self.base_url;
-
-        let mut new_session_url = base
-            .join("newSession")
-            .map_err(|e| Error::ConfigError(e.to_string()))?;
-
-        let mut q_params = String::new();
-
-        match nonce {
-            Nonce::Value(val) if !val.is_empty() => {
-                q_params.push_str("nonce=");
-                q_params.push_str(&base64::encode_config(val, base64::URL_SAFE));
-            }
-            Nonce::Size(val) if *val > 0 => {
-                q_params.push_str("nonceSize=");
-                q_params.push_str(&val.to_string());
-            }
-            _ => {}
-        }
-
-        new_session_url.set_query(Some(&q_params));
-
-        Ok(new_session_url)
-    }
-
-    fn challenge_response(
+    /// Execute a challenge/response operation with the given evidence.
+    pub fn challenge_response(
```

Comment:
In the long term this method will either need more detailed documentation (especially around what the various params need to be), or a refactoring of its interface into something that prevents "wrong" inputs to be passed in; (this is just a note, not a request for a change)